scipy/linalg/tests/test_lapack.py

```diff
@@ -1654,3 +1654,158 @@ def test_getc2_gesc2():
         else:
             assert_array_almost_equal(desired_cplx.astype(dtype),
                                       x/scale, decimal=4)
+
+
+def generate_random_dtype_array(shape, dtype):
+    # generates a random matrix of desired data type of shape
+    if dtype in COMPLEX_DTYPES:
+        return (np.random.rand(*shape)
+                + np.random.rand(*shape)*1.0j).astype(dtype)
+    return np.random.rand(*shape).astype(dtype)
+
+
+@pytest.mark.parametrize("size", [(6, 5), (5, 5)])
+@pytest.mark.parametrize('dtype', DTYPES)
+@pytest.mark.parametrize('gejsv_lambda',
+                         [(lambda A, gejsv: gejsv(A)),  # base default case
+                          (lambda A, gejsv: gejsv(A, joba="E")),  # joba
+                          (lambda A, gejsv: gejsv(A, joba="F")),
+                          (lambda A, gejsv: gejsv(A, joba="G")),
+                          (lambda A, gejsv: gejsv(A, joba="A")),
+                          (lambda A, gejsv: gejsv(A, joba="R")),
+                          (lambda A, gejsv: gejsv(A, jobu="F")),  # jobu
+                          (lambda A, gejsv: gejsv(A, jobu="W")),
+                          (lambda A, gejsv: gejsv(A, jobv="J")),  # jobv
+                          (lambda A, gejsv: gejsv(A, jobv="V")),
+                          (lambda A, gejsv: gejsv(A, jobr="N")),  # jobr
+                          (lambda A, gejsv: gejsv(A, jobt="T")),  # jobt
+                          (lambda A, gejsv: gejsv(A, jobp="N")),  # jobp
+                          ])
+def test_gejsv_general(size, dtype, gejsv_lambda):
+    '''
+    Tests the lapack routine ?gejsv. This function varies the job? values
+    one at a time per dtype by passing different dtype routines into a lambda
+    that decides parameters. It then tests the general random matrix A of size
+    (m, n) for zero info, that sva, v, and u can be recombined to make A, and
+    that u.T @ u and v.T @ v are the identity. Lastly, we check that iwork [0]
+    and [1] are n.
+    '''
+    np.random.seed(42)
```

Comment:
```suggestion
    seed(42)
```
@ilayn has commented before that `seed` has absolute import

Reply:
Done.